### ISSUE TITLE
SC2039 wiki is incomplete

### ISSUE THREAD

**Issue by zapashcanon** [koalaman/shellcheck#917]
Hi,

I don't know if this is because there are a lot of warnings of this kind so it's not here by choice, but I just saw a warning which isn't explained on the wiki:

> SC2039: In POSIX sh, [[ ]] is undefined.

A note about it would be great.

**Comment by Nightfirecat** [comment]
> You have declared that your script works with `/bin/sh`, but you are using features that have undefined behavior according to the POSIX specification.
>
> Either declare that your script requires a specific shell like `#!/bin/bash` or `#!/bin/dash`, or rewrite the script in a portable way.

Basically, `sh` does not support double-bracket expressions; that is a feature of `bash`/`dash`.

**Comment by zapashcanon** [comment]
I know, but in most of the other wiki pages, there a little note + a way to fix this (not by moving to another shell of course..).

**Comment by Nightfirecat** [comment]
Well, there's lots of things that you *could* use that aren't POSIX compatible... Given that that particular page is meant to be generic to any non-POSIX feature, I don't know of a good way for it to suggest a substitution.

**Comment by koalaman** [comment]
Agreed. This warning should be a whole heap of different ones instead, each with its own wiki page. See #503

**Comment by MYDavoodeh** [comment]
I found another instance that I'd like to share here:
```sh
#!/bin/sh
magic=var
var=value
[ -n "${!magic}" ] && echo "something"
       ^--^ SC2039: In POSIX sh, indirect expansion is undefined.
```
Throws the error, but there is not much of explaination related to it (which is fine but not so noob friendly).

In case you are here to find the answer: you can fix it by replacing `"${!magic}"` with `"$(eval printf "%s" "\$$magic")"` (prints "$var" and gets the "value")
